gradio/blocks.py

```diff
@@ -843,18 +775,30 @@ async def call_function(
                     for input_component, data in zip(block_fn.inputs, processed_input)
                 }
             ]
-        processed_input = add_request_to_inputs(
-            block_fn.fn, list(processed_input), request
+
+        processed_input, _, progress_index = special_args(
+            block_fn, processed_input, request
+        )
+        progress_tracker = (
+            processed_input[progress_index] if progress_index is not None else None
         )
 
         start = time.time()
 
         if iterator is None:  # If not a generator function that has already run
-            if inspect.iscoroutinefunction(block_fn.fn):
-                prediction = await block_fn.fn(*processed_input)
+            if progress_tracker is not None:
+                progress_tracker, fn = create_tracker(
+                    self, event_id, block_fn.fn, progress_tracker.track_tqdm
```

Comment:
I think it will be cleaner if the `event_id` is not passed to the Blocks instance and instead the progress tracker is created in the route, similar to how `Requests` are created and then passed to the function